src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -527,40 +527,42 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
 
 
   private def validateClusterRequestBucketObjectUri(userEmail: WorkbenchEmail, googleProject: GoogleProject, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[Unit] = {
-    for {
+    val transformed = for {
+      // Get a pet token from Sam. If we can't get a token, we won't do validation but won't fail cluster creation.
+      petToken <- OptionT(serviceAccountProvider.getAccessToken(userEmail, googleProject))
+
+      // Validate the user script URI
       _ <- clusterRequest.jupyterUserScriptUri match {
-        case Some(userScriptUri) => validateBucketObjectUri(userEmail, googleProject, userScriptUri.toUri)
-        case None => Future.successful(())
+        case Some(userScriptUri) => OptionT.liftF[Future, Unit](validateBucketObjectUri(userEmail, petToken, googleProject, userScriptUri.toUri))
+        case None => OptionT.pure[Future, Unit](())
       }
 
+      // Validate the extension URIs
       _ <- clusterRequest.userJupyterExtensionConfig match {
         case Some(config) =>
           val extensionsToValidate = (config.nbExtensions.values ++ config.serverExtensions.values ++ config.combinedExtensions.values).filter(_.startsWith("gs://"))
-          Future.traverse(extensionsToValidate)(x => validateBucketObjectUri(userEmail, googleProject, x))
-        case None => Future.successful(())
+          OptionT.liftF(Future.traverse(extensionsToValidate)(x => validateBucketObjectUri(userEmail, petToken, googleProject, x)))
+        case None => OptionT.pure[Future, Unit](())
       }
-    } yield()
+    } yield ()
+
+    transformed.value.void
   }
-  private[service] def validateBucketObjectUri(userEmail: WorkbenchEmail, googleProject: GoogleProject, gcsUri: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
 
+  private[service] def validateBucketObjectUri(userEmail: WorkbenchEmail, userToken: String, googleProject: GoogleProject, gcsUri: String)(implicit executionContext: ExecutionContext): Future[Unit] = {
     val gcsUriOpt = parseGcsPath(gcsUri)
     gcsUriOpt match {
+      case Left(_) => Future.failed(BucketObjectException(gcsUri))
+      case Right(gcsPath) if gcsPath.toUri.length > bucketPathMaxLength => Future.failed(BucketObjectException(gcsUri))
       case Right(gcsPath) =>
-        if (gcsPath.toUri.length > bucketPathMaxLength)
```

Comment:
This `if` was a useless expression - there probably should have been an `else`? Changed it to use pattern matching anyway.